app/services/PenaltiesService.scala

```diff
@@ -182,6 +181,8 @@ class PenaltiesService @Inject()(config: FrontendAppConfig,
         .filter(penaltyPstr => !schemesWithPstr.map(_.pstr.get).contains(penaltyPstr))
         .map(x => PenaltySchemes(None, x, None))
 
+      fiCacheConnector.save(Json.obj("pstrs" -> unassociatedSchemes.map(_.pstr)))
```

Comment:
You need to map the future to the following line, else it's possible that it might fail or not complete in time.

Reply:
🤦  schoolboy error. Updating..

Reply:
Updated